src/cdsetool/download.py

```diff
@@ -126,3 +143,60 @@ def _set_proxy(options, session):
     if proxies != {}:
         session.proxies.update(proxies)
     return session
+
+
+def validity_check(temp_path, product_info):
+    """Given it's temp_path and metadata info checks if the data downloaded is valid"""
+    size = os.path.getsize(temp_path)
+    content_length = (
+        product_info.get("properties").get("services").get("download").get("size")
+    )
+    if size > content_length:
+        # Dowloaded more than the size of the file.
+        return Validity.INVALID
+    if size == content_length:
```

Comment:
If you change the condition to `if size != content_length: return Validity.CONTINUE` you can save a nesting level for the following code.

Reply:
I guess your suggesting would point out that we don't want to consider to restart downloading partial files (the `Validity.Continue` I inteded to use it just in case of restarting partial files and not for corrupted ones)?
for example we download the first 5000 Bytes then by checking the validiy I used `Validity.Continue` for retry the download, but this part is still am Idea.

Reply:
My suggestion doesn't change the logic as far as I can tell:
```python
if size != content_length:
    return Validity.CONTINUE
checksum_comparison = _checksum_compare(temp_path, product_info)
if checksum_comparison is None:
  # No data available for the checksum
  return Validity.IGNORE
if checksum_comparison:
  return Validity.VALID
# Checksum failed
return Validity.INVALID
```

Reply:
The problem with using 
```Python
if size != content_length:
    return Validity.CONTINUE
``` 
is that you enter that condition two times if you donloaded more of the actual size or less, which are two different logics. If you downloaded more than the file is corrupted if you donwloaded less then you can resume the donwload (but this logic is yet to implement or if we want to implement as suggested in [Restart a partial download)](https://github.com/SDFIdk/CDSETool/issues/69)

```Python 
    if size > content_length:
        return Validity.INVALID
    if size == content_length:
        return _checksum_compare(temp_path, product_info)
    # here it's a partial download logic to implement is TODO
    return Validity.CONTINUE
```

current implementation as suggested in 

> I think changing line 178 to `return Validity.IGNORE` and this line to `return Validity.VALID if algo.hexdigest().lower() == checksum.lower() else Validity.INVALID` would eliminate the need for the logic on lines 159-165.

Reply:
Sorry, I don't understand what you are saying. Which part of this doesn't work?
```python
if size > content_length:
  return Validity.INVALID
if size != content_length:
  # here it's a partial download logic to implement is TODO
  return Validity.CONTINUE
return _checksum_compare(temp_path, product_info)
```
And seeing that code spelled out, the condition should probably be `<` rather than `!=`.

Reply:
AH ok I see I was missing some parts before this one works for me I will add now